doc/EXAM.md

```diff
@@ -0,0 +1,157 @@
+
+	* `can_start`
+		When the exam is initially planned to begin, e.g. 12.00.
+
+	* `start_time_actual`
+		When the exam actually started, e.g. 12.04.
+
+* ExamAttempt
+
+	`ExamAttempt` class is responsible for a single exam attempt by a single student.
+
+	* `exam_taken`
+		The `ExamSession` that the student is attempting.
+
+	* `student`
+		The `UserProfile` that is taking the exam.
+
+	* `exam_started`
+		The time when the student started the exam.
+
+	* `exam_finished`
+		When student finishes the exam.
+
+	* `system_identifier`
+		The indentifier for the machine that is used to do the exam.
+
+
+### Views
+
+Many views are same as in `exercise` views, but a few are added for new detail pages for the exam.
+
+* ExamStartView
+
+	This view is initially shown to the student and it shows a list of all available exams to the student.
+	By clicking on the correct exam, it will direct student to `ExamDetailView`. Uses the template
+	`exam_start.html`.
+
+* ExamDetailView
+
+	Shows details of a single exam. It includes all the details that are unique for that specific exam.
+	Uses the template `exam_detail.html`. The view is able to deduce the template name without `template_name`
+	parameter, this is a feature in Django.
+
+
+### Templates
+
+The templates have had many changes, since in exam mode we need to disable many links
+and the UI should be as simple as possible.
+
+* `exammode/base.html`
+
+	This is the base template for exammode and it is a custom version of `templates/base.html`.
+	This is set in 'exercises/views.py'file. The main modifications include removing the user menu on the left side of the page
+	and displaying just the language selection, student ID and user name on the page header.
+
+* `exammode/examsession_detail.html`
+
+	Used by the `ExamDetailView`.
+
+* `exammode/exam_start.html`
+
+	Used by the `ExamStartView`.
+
+* `exammode/exam.html`
+
+	This is the exam template that is shown when the student is on the exam page.
+	The template is created by combining `exercise/exercise_base.html` and
+	`exercise/exercise.html` together and making necessary modifications for exam page.
+	The reason to combine them was to make the templates a bit more readable, since in exam the templates are simpler.
+	So it is not necessary to split them like they are in `exercise`.
+
+* `exammode/exam_question.html`
+
+	This is the template for a single question. The questions are wrapped in Bootstrap panels and this is wrapping is done in this template.
+	The template is a custom version of `exercise/exercise_plain.html`.
+	The exercise is also split into two columns in the exam version.
+	The splitting is achieved in `static/exammode/js/exam.js` with the help of Bootstrap's grid system.
+
+
+Exam Management Page
+--------------------
+
+Exam management page was added to A+ to create, edit and delete ExamSessions.
+It is found in course menu by the course staff.
+
+### Additions to `exercise/staff_views.py`
+
+* ExamManagementView
```

Comment:
Is this document up-to-date? Some of these were supposed to be moved to the exammode app.

Reply:
Fixed documentation